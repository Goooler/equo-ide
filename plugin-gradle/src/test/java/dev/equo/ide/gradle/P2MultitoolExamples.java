/*******************************************************************************
 * Copyright (c) 2022 EquoTech, Inc. and others.
 *
 * This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License 2.0
 * which accompanies this distribution, and is available at
 * https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *     EquoTech, Inc. - initial API and implementation
 *******************************************************************************/
package dev.equo.ide.gradle;

import au.com.origin.snapshots.Expect;
import au.com.origin.snapshots.junit5.SnapshotExtension;
import java.io.IOException;
import org.junit.jupiter.api.Disabled;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;

@ExtendWith({SnapshotExtension.class})
@Disabled
public class P2MultitoolExamples extends GradleHarness {
	@Test
	public void _01(Expect expect) throws IOException {
		setFile("build.gradle")
				.toLines(
						"plugins { id 'dev.equo.ide' }",
						"equoIde {",
						"  p2repo 'https://download.eclipse.org/eclipse/updates/4.26/'",
						"}");
		run("-q", "equoList", "--all=categories").snapshot(expect);
	}

	@Test
	public void _02(Expect expect) throws IOException {
		setFile("build.gradle")
				.toLines(
						"plugins { id 'dev.equo.ide' }",
						"equoIde {",
						"  p2repo 'https://download.eclipse.org/eclipse/updates/4.26/'",
						"}");
		run("-q", "equoList", "--installed").snapshot(expect);
	}

	@Test
	public void _03(Expect expect) throws IOException {
		setFile("build.gradle")
				.toLines(
						"plugins { id 'dev.equo.ide' }",
						"equoIde {",
						"  p2repo 'https://download.eclipse.org/eclipse/updates/4.26/'",
						"  install 'org.eclipse.releng.java.languages.categoryIU'",
						"}");
		run("-q", "equoList", "--installed").snapshot(expect);
	}

	@Test
	public void _04(Expect expect) throws IOException {
		setFile("build.gradle")
				.toLines(
						"plugins { id 'dev.equo.ide' }",
						"equoIde {",
						"  p2repo 'https://download.eclipse.org/eclipse/updates/4.26/'",
						"  install 'org.eclipse.releng.java.languages.categoryIU'",
						"}");
		run("-q", "equoList", "--problems").snapshot(expect);
	}

	@Test
	public void _05(Expect expect) throws IOException {
		setFile("build.gradle")
				.toLines(
						"plugins { id 'dev.equo.ide' }",
						"equoIde {",
						"  p2repo 'https://download.eclipse.org/eclipse/updates/4.26/'",
						"  install 'org.eclipse.releng.java.languages.categoryIU'",
						"}");
		run("-q", "equoList", "--detail=org.eclipse.jdt.compiler.apt", "--stacktrace")
				.snapshot(expect.scenario("apt"));
		run("-q", "equoList", "--detail=org.eclipse.jdt.core.compiler.batch", "--stacktrace")
				.snapshot(expect.scenario("batch"));
	}

	@Test
	public void _06(Expect expect) throws IOException {
		setFile("build.gradle")
				.toLines(
						"plugins { id 'dev.equo.ide' }",
						"equoIde {",
						"  p2repo 'https://download.eclipse.org/eclipse/updates/4.26/'",
						"  install 'org.eclipse.releng.java.languages.categoryIU'",
						"}");
		run("-q", "equoList", "--raw=org.eclipse.jdt.core.compiler.batch", "--stacktrace")
				.snapshot(expect.scenario("batch"));
	}
}
