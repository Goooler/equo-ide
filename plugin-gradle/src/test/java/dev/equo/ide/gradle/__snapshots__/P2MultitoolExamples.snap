dev.equo.ide.gradle.P2MultitoolExamples._01=[
+----------------------------------------------+---------------------------------------------------+
| id                                           | name \n description                               |
+----------------------------------------------+---------------------------------------------------+
| I20221123-1800.Default                       | Uncategorized                                     |
|                                              |   Default category for otherwise uncategorized    |
|                                              |   features                                        |
| org.eclipse.equinox.target.categoryIU        | Equinox Target Components                         |
|                                              |   Features especially useful to install as PDE    |
|                                              |   runtime targets.                                |
| org.eclipse.platform.ide.categoryIU          | Eclipse Platform                                  |
|                                              |   Minimum version of Eclipse: no source or API    |
|                                              |   documentation, no PDE or JDT.                   |
| org.eclipse.platform.sdk.categoryIU          | Eclipse Platform SDK                              |
|                                              |   Minimum version of Eclipse with source and      |
|                                              |   documentation, no PDE or JDT.                   |
| org.eclipse.rcp.categoryIU                   | Eclipse RCP Target Components                     |
|                                              |   Features to use as PDE runtime target, while    |
|                                              |   developing RCP applications.                    |
| org.eclipse.releng.java.languages.categoryIU | Eclipse Java Development Tools                    |
|                                              |   Tools to allow development with Java.           |
| org.eclipse.releng.pde.categoryIU            | Eclipse Plugin Development Tools                  |
|                                              |   Tools to develop bundles, plugins and features. |
| org.eclipse.releng.testsIU                   | Eclipse Tests, Tools, Examples, and Extras        |
|                                              |   Collection of Misc. Features, such as unit      |
|                                              |   tests, SWT and e4 tools, examples, and          |
|                                              |   compatibility features not shipped as part of   |
|                                              |   main SDK, but which some people may desire in   |
|                                              |   creating products based on previous versions of |
|                                              |   Eclipse.                                        |
| org.eclipse.sdk.ide.categoryIU               | Eclipse SDK                                       |
|                                              |   The full version of Eclipse, with source and    |
|                                              |   documentation: Platform, JDT and PDE.           |
+----------------------------------------------+---------------------------------------------------+
]


dev.equo.ide.gradle.P2MultitoolExamples._02=[
No jars were specified.
]


dev.equo.ide.gradle.P2MultitoolExamples._03=[
WARNING!!! 0 unmet requirement(s), 7 ambigous requirement(s). For more info:
WARNING!!! gradlew equoList --problems
+---------------------------------------------------------------------------------+------------------------+
| maven coordinate / p2 id                                                        | repo                   |
+---------------------------------------------------------------------------------+------------------------+
| com.ibm.icu:icu4j:72.1                                                          | maven central          |
| commons-io:commons-io:2.11.0                                                    | maven central          |
| net.java.dev.jna:jna-platform:5.12.1                                            | maven central          |
| net.java.dev.jna:jna:5.12.1                                                     | maven central          |
| org.apiguardian:apiguardian-api:1.1.2                                           | maven central          |
| org.apiguardian:apiguardian-api:1.1.2                                           | maven central          |
| org.bouncycastle:bcpg-jdk18on:1.72                                              | maven central          |
| org.bouncycastle:bcprov-jdk18on:1.72                                            | maven central          |
| org.junit.jupiter:junit-jupiter-api:5.9.1                                       | maven central          |
| org.junit.jupiter:junit-jupiter-api:5.9.1                                       | maven central          |
| org.junit.jupiter:junit-jupiter-engine:5.9.1                                    | maven central          |
| org.junit.jupiter:junit-jupiter-engine:5.9.1                                    | maven central          |
| org.junit.jupiter:junit-jupiter-migrationsupport:5.9.1                          | maven central          |
| org.junit.jupiter:junit-jupiter-migrationsupport:5.9.1                          | maven central          |
| org.junit.jupiter:junit-jupiter-params:5.9.1                                    | maven central          |
| org.junit.jupiter:junit-jupiter-params:5.9.1                                    | maven central          |
| org.junit.platform:junit-platform-commons:1.9.1                                 | maven central          |
| org.junit.platform:junit-platform-commons:1.9.1                                 | maven central          |
| org.junit.platform:junit-platform-engine:1.9.1                                  | maven central          |
| org.junit.platform:junit-platform-engine:1.9.1                                  | maven central          |
| org.junit.platform:junit-platform-launcher:1.9.1                                | maven central          |
| org.junit.platform:junit-platform-launcher:1.9.1                                | maven central          |
| org.junit.platform:junit-platform-runner:1.9.1                                  | maven central          |
| org.junit.platform:junit-platform-runner:1.9.1                                  | maven central          |
| org.junit.platform:junit-platform-suite-api:1.9.1                               | maven central          |
| org.junit.platform:junit-platform-suite-api:1.9.1                               | maven central          |
| org.junit.platform:junit-platform-suite-commons:1.9.1                           | maven central          |
| org.junit.platform:junit-platform-suite-commons:1.9.1                           | maven central          |
| org.junit.platform:junit-platform-suite-engine:1.9.1                            | maven central          |
| org.junit.platform:junit-platform-suite-engine:1.9.1                            | maven central          |
| org.junit.vintage:junit-vintage-engine:5.9.1                                    | maven central          |
| org.junit.vintage:junit-vintage-engine:5.9.1                                    | maven central          |
| org.opentest4j:opentest4j:1.2.0                                                 | maven central          |
| org.opentest4j:opentest4j:1.2.0                                                 | maven central          |
| org.osgi:org.osgi.service.event:1.4.1                                           | maven central          |
| org.osgi:org.osgi.service.prefs:1.1.2                                           | maven central          |
| org.tukaani:xz:1.9                                                              | maven central          |
| org.eclipse.emf:org.eclipse.emf.common:2.27.0                                   | maven central?         |
| org.eclipse.emf:org.eclipse.emf.ecore.change:2.14.0                             | maven central?         |
| org.eclipse.emf:org.eclipse.emf.ecore.xmi:2.17.0                                | maven central?         |
| org.eclipse.emf:org.eclipse.emf.ecore:2.29.0                                    | maven central?         |
| org.eclipse.jdt:org.eclipse.jdt.annotation:2.2.700                              | maven central?         |
| org.eclipse.jdt:org.eclipse.jdt.apt.core:3.7.50                                 | maven central?         |
| org.eclipse.jdt:org.eclipse.jdt.apt.pluggable.core:1.3.0                        | maven central?         |
| org.eclipse.jdt:org.eclipse.jdt.apt.ui:3.7.0                                    | maven central?         |
| org.eclipse.jdt:org.eclipse.jdt.compiler.apt:1.4.300                            | maven central?         |
| org.eclipse.jdt:org.eclipse.jdt.compiler.tool:1.3.200                           | maven central?         |
| org.eclipse.jdt:org.eclipse.jdt.core.formatterapp:1.1.0                         | maven central?         |
| org.eclipse.jdt:org.eclipse.jdt.core.manipulation:1.17.0                        | maven central?         |
| org.eclipse.jdt:org.eclipse.jdt.core:3.32.0                                     | maven central?         |
| org.eclipse.jdt:org.eclipse.jdt.debug.ui:3.12.900                               | maven central?         |
| org.eclipse.jdt:org.eclipse.jdt.debug:3.20.0                                    | maven central?         |
| org.eclipse.jdt:org.eclipse.jdt.doc.isv:3.14.1800                               | maven central?         |
| org.eclipse.jdt:org.eclipse.jdt.doc.user:3.15.1600                              | maven central?         |
| org.eclipse.jdt:org.eclipse.jdt.junit.core:3.11.500                             | maven central?         |
| org.eclipse.jdt:org.eclipse.jdt.junit.runtime:3.7.0                             | maven central?         |
| org.eclipse.jdt:org.eclipse.jdt.junit4.runtime:1.3.0                            | maven central?         |
| org.eclipse.jdt:org.eclipse.jdt.junit5.runtime:1.1.100                          | maven central?         |
| org.eclipse.jdt:org.eclipse.jdt.junit:3.15.100                                  | maven central?         |
| org.eclipse.jdt:org.eclipse.jdt.launching:3.19.800                              | maven central?         |
| org.eclipse.jdt:org.eclipse.jdt.ui:3.27.100                                     | maven central?         |
| org.eclipse.jdt:org.eclipse.jdt:3.18.1400                                       | maven central?         |
| org.eclipse.platform:org.eclipse.ant.core:3.6.500                               | maven central?         |
| org.eclipse.platform:org.eclipse.ant.launching:1.3.400                          | maven central?         |
| org.eclipse.platform:org.eclipse.ant.ui:3.8.300                                 | maven central?         |
| org.eclipse.platform:org.eclipse.compare.core:3.7.100                           | maven central?         |
| org.eclipse.platform:org.eclipse.compare:3.8.500                                | maven central?         |
| org.eclipse.platform:org.eclipse.core.commands:3.10.300                         | maven central?         |
| org.eclipse.platform:org.eclipse.core.contenttype:3.8.200                       | maven central?         |
| org.eclipse.platform:org.eclipse.core.databinding.observable:1.12.100           | maven central?         |
| org.eclipse.platform:org.eclipse.core.databinding.property:1.9.100              | maven central?         |
| org.eclipse.platform:org.eclipse.core.databinding:1.11.200                      | maven central?         |
| org.eclipse.platform:org.eclipse.core.expressions:3.8.200                       | maven central?         |
| org.eclipse.platform:org.eclipse.core.externaltools:1.2.300                     | maven central?         |
| org.eclipse.platform:org.eclipse.core.filebuffers:3.7.200                       | maven central?         |
| org.eclipse.platform:org.eclipse.core.filesystem:1.9.500                        | maven central?         |
| org.eclipse.platform:org.eclipse.core.jobs:3.13.200                             | maven central?         |
| org.eclipse.platform:org.eclipse.core.resources:3.18.100                        | maven central?         |
| org.eclipse.platform:org.eclipse.core.runtime:3.26.100                          | maven central?         |
| org.eclipse.platform:org.eclipse.core.variables:3.5.100                         | maven central?         |
| org.eclipse.platform:org.eclipse.debug.core:3.20.0                              | maven central?         |
| org.eclipse.platform:org.eclipse.debug.ui:3.17.100                              | maven central?         |
| org.eclipse.platform:org.eclipse.e4.core.commands:1.0.300                       | maven central?         |
| org.eclipse.platform:org.eclipse.e4.core.contexts:1.11.0                        | maven central?         |
| org.eclipse.platform:org.eclipse.e4.core.di.annotations:1.7.200                 | maven central?         |
| org.eclipse.platform:org.eclipse.e4.core.di.extensions.supplier:0.16.400        | maven central?         |
| org.eclipse.platform:org.eclipse.e4.core.di.extensions:0.17.200                 | maven central?         |
| org.eclipse.platform:org.eclipse.e4.core.di:1.8.300                             | maven central?         |
| org.eclipse.platform:org.eclipse.e4.core.services:2.3.400                       | maven central?         |
| org.eclipse.platform:org.eclipse.e4.emf.xpath:0.3.100                           | maven central?         |
| org.eclipse.platform:org.eclipse.e4.ui.bindings:0.13.200                        | maven central?         |
| org.eclipse.platform:org.eclipse.e4.ui.css.core:0.13.400                        | maven central?         |
| org.eclipse.platform:org.eclipse.e4.ui.css.swt.theme:0.13.200                   | maven central?         |
| org.eclipse.platform:org.eclipse.e4.ui.css.swt:0.14.700                         | maven central?         |
| org.eclipse.platform:org.eclipse.e4.ui.di:1.4.100                               | maven central?         |
| org.eclipse.platform:org.eclipse.e4.ui.dialogs:1.3.400                          | maven central?         |
| org.eclipse.platform:org.eclipse.e4.ui.ide:3.16.200                             | maven central?         |
| org.eclipse.platform:org.eclipse.e4.ui.model.workbench:2.2.300                  | maven central?         |
| org.eclipse.platform:org.eclipse.e4.ui.services:1.5.100                         | maven central?         |
| org.eclipse.platform:org.eclipse.e4.ui.widgets:1.3.100                          | maven central?         |
| org.eclipse.platform:org.eclipse.e4.ui.workbench.addons.swt:1.4.500             | maven central?         |
| org.eclipse.platform:org.eclipse.e4.ui.workbench.renderers.swt:0.15.700         | maven central?         |
| org.eclipse.platform:org.eclipse.e4.ui.workbench.swt:0.16.700                   | maven central?         |
| org.eclipse.platform:org.eclipse.e4.ui.workbench3:0.16.100                      | maven central?         |
| org.eclipse.platform:org.eclipse.e4.ui.workbench:1.14.0                         | maven central?         |
| org.eclipse.platform:org.eclipse.equinox.app:1.6.200                            | maven central?         |
| org.eclipse.platform:org.eclipse.equinox.bidi:1.4.200                           | maven central?         |
| org.eclipse.platform:org.eclipse.equinox.common:3.17.0                          | maven central?         |
| org.eclipse.platform:org.eclipse.equinox.frameworkadmin.equinox:1.2.200         | maven central?         |
| org.eclipse.platform:org.eclipse.equinox.frameworkadmin:2.2.100                 | maven central?         |
| org.eclipse.platform:org.eclipse.equinox.p2.artifact.repository:1.4.600         | maven central?         |
| org.eclipse.platform:org.eclipse.equinox.p2.core:2.9.200                        | maven central?         |
| org.eclipse.platform:org.eclipse.equinox.p2.engine:2.7.500                      | maven central?         |
| org.eclipse.platform:org.eclipse.equinox.p2.jarprocessor:1.2.300                | maven central?         |
| org.eclipse.platform:org.eclipse.equinox.p2.metadata.repository:1.4.100         | maven central?         |
| org.eclipse.platform:org.eclipse.equinox.p2.metadata:2.6.300                    | maven central?         |
| org.eclipse.platform:org.eclipse.equinox.p2.repository:2.6.300                  | maven central?         |
| org.eclipse.platform:org.eclipse.equinox.preferences:3.10.100                   | maven central?         |
| org.eclipse.platform:org.eclipse.equinox.registry:3.11.200                      | maven central?         |
| org.eclipse.platform:org.eclipse.equinox.security:1.3.1000                      | maven central?         |
| org.eclipse.platform:org.eclipse.equinox.simpleconfigurator.manipulator:2.2.100 | maven central?         |
| org.eclipse.platform:org.eclipse.equinox.simpleconfigurator:1.4.200             | maven central?         |
| org.eclipse.platform:org.eclipse.equinox.supplement:1.10.600                    | maven central?         |
| org.eclipse.platform:org.eclipse.help:3.9.100                                   | maven central?         |
| org.eclipse.platform:org.eclipse.jface.databinding:1.14.0                       | maven central?         |
| org.eclipse.platform:org.eclipse.jface.text:3.22.0                              | maven central?         |
| org.eclipse.platform:org.eclipse.jface:3.28.0                                   | maven central?         |
| org.eclipse.platform:org.eclipse.ltk.core.refactoring:3.13.0                    | maven central?         |
| org.eclipse.platform:org.eclipse.ltk.ui.refactoring:3.12.200                    | maven central?         |
| org.eclipse.platform:org.eclipse.osgi:3.18.200                                  | maven central?         |
| org.eclipse.platform:org.eclipse.search:3.14.300                                | maven central?         |
| org.eclipse.platform:org.eclipse.swt:3.122.0                                    | maven central?         |
| org.eclipse.platform:org.eclipse.team.core:3.9.600                              | maven central?         |
| org.eclipse.platform:org.eclipse.team.ui:3.9.500                                | maven central?         |
| org.eclipse.platform:org.eclipse.text:3.12.300                                  | maven central?         |
| org.eclipse.platform:org.eclipse.ui.cheatsheets:3.7.500                         | maven central?         |
| org.eclipse.platform:org.eclipse.ui.console:3.11.400                            | maven central?         |
| org.eclipse.platform:org.eclipse.ui.editors:3.14.400                            | maven central?         |
| org.eclipse.platform:org.eclipse.ui.externaltools:3.5.200                       | maven central?         |
| org.eclipse.platform:org.eclipse.ui.forms:3.11.500                              | maven central?         |
| org.eclipse.platform:org.eclipse.ui.ide:3.20.0                                  | maven central?         |
| org.eclipse.platform:org.eclipse.ui.intro:3.6.600                               | maven central?         |
| org.eclipse.platform:org.eclipse.ui.navigator.resources:3.8.500                 | maven central?         |
| org.eclipse.platform:org.eclipse.ui.navigator:3.10.400                          | maven central?         |
| org.eclipse.platform:org.eclipse.ui.views.properties.tabbed:3.9.300             | maven central?         |
| org.eclipse.platform:org.eclipse.ui.views:3.11.300                              | maven central?         |
| org.eclipse.platform:org.eclipse.ui.workbench.texteditor:3.16.600               | maven central?         |
| org.eclipse.platform:org.eclipse.ui.workbench:3.127.0                           | maven central?         |
| org.eclipse.platform:org.eclipse.ui:3.201.200                                   | maven central?         |
| org.eclipse.platform:org.eclipse.urischeme:1.2.200                              | maven central?         |
| javax.annotation:1.3.5.v20221112-0806                                           | p2 R-4.26-202211231800 |
| javax.inject:1.0.0.v20220405-0441                                               | p2 R-4.26-202211231800 |
| org.apache.ant:1.10.12.v20211102-1452                                           | p2 R-4.26-202211231800 |
| org.apache.batik.constants:1.16.0.v20221027-0840                                | p2 R-4.26-202211231800 |
| org.apache.batik.css:1.16.0.v20221027-0840                                      | p2 R-4.26-202211231800 |
| org.apache.batik.i18n:1.16.0.v20221027-0840                                     | p2 R-4.26-202211231800 |
| org.apache.batik.util:1.16.0.v20221027-0840                                     | p2 R-4.26-202211231800 |
| org.apache.commons.jxpath:1.3.0.v200911051830                                   | p2 R-4.26-202211231800 |
| org.apache.commons.logging:1.2.0.v20180409-1502                                 | p2 R-4.26-202211231800 |
| org.apache.xmlgraphics:2.7.0.v20221018-0736                                     | p2 R-4.26-202211231800 |
| org.eclipse.ant.launching.source:1.3.400.v20220718-1722                         | p2 R-4.26-202211231800 |
| org.eclipse.ant.ui.source:3.8.300.v20220718-1722                                | p2 R-4.26-202211231800 |
| org.eclipse.jdt.annotation.source:2.2.700.v20220826-1026                        | p2 R-4.26-202211231800 |
| org.eclipse.jdt.apt.core.source:3.7.50.v20210914-1429                           | p2 R-4.26-202211231800 |
| org.eclipse.jdt.apt.pluggable.core.source:1.3.0.v20210618-1653                  | p2 R-4.26-202211231800 |
| org.eclipse.jdt.apt.ui.source:3.7.0.v20210620-1751                              | p2 R-4.26-202211231800 |
| org.eclipse.jdt.compiler.apt.source:1.4.300.v20221108-0856                      | p2 R-4.26-202211231800 |
| org.eclipse.jdt.compiler.tool.source:1.3.200.v20220802-0458                     | p2 R-4.26-202211231800 |
| org.eclipse.jdt.core.formatterapp.source:1.1.0.v20210618-1653                   | p2 R-4.26-202211231800 |
| org.eclipse.jdt.core.manipulation.source:1.17.0.v20221026-1918                  | p2 R-4.26-202211231800 |
| org.eclipse.jdt.core.source:3.32.0.v20221108-1853                               | p2 R-4.26-202211231800 |
| org.eclipse.jdt.debug.source:3.20.0.v20220922-0905                              | p2 R-4.26-202211231800 |
| org.eclipse.jdt.debug.ui.source:3.12.900.v20221001-0715                         | p2 R-4.26-202211231800 |
| org.eclipse.jdt.junit.core.source:3.11.500.v20221031-1935                       | p2 R-4.26-202211231800 |
| org.eclipse.jdt.junit.runtime.source:3.7.0.v20220609-1843                       | p2 R-4.26-202211231800 |
| org.eclipse.jdt.junit.source:3.15.100.v20220909-2154                            | p2 R-4.26-202211231800 |
| org.eclipse.jdt.junit4.runtime.source:1.3.0.v20220609-1843                      | p2 R-4.26-202211231800 |
| org.eclipse.jdt.junit5.runtime.source:1.1.100.v20220907-0450                    | p2 R-4.26-202211231800 |
| org.eclipse.jdt.launching.macosx.source:3.5.100.v20220927-0854                  | p2 R-4.26-202211231800 |
| org.eclipse.jdt.launching.source:3.19.800.v20221107-1851                        | p2 R-4.26-202211231800 |
| org.eclipse.jdt.launching.ui.macosx.source:1.3.100.v20220927-0854               | p2 R-4.26-202211231800 |
| org.eclipse.jdt.ui.source:3.27.100.v20221122-0749                               | p2 R-4.26-202211231800 |
| org.hamcrest.core.source:1.3.0.v20180420-1519                                   | p2 R-4.26-202211231800 |
| org.hamcrest.core:1.3.0.v20180420-1519                                          | p2 R-4.26-202211231800 |
| org.junit.source:4.13.2.v20211018-1956                                          | p2 R-4.26-202211231800 |
| org.junit:4.13.2.v20211018-1956                                                 | p2 R-4.26-202211231800 |
| org.w3c.css.sac:1.3.1.v200903091627                                             | p2 R-4.26-202211231800 |
| org.w3c.dom.smil:1.0.1.v200903091627                                            | p2 R-4.26-202211231800 |
| org.w3c.dom.svg:1.1.0.v201011041433                                             | p2 R-4.26-202211231800 |
+---------------------------------------------------------------------------------+------------------------+
]


dev.equo.ide.gradle.P2MultitoolExamples._04=[
No unmet requirements.
+-------------------------------------------------------------+-----------------------------------------------------------+-----------+
| ambiguous requirement                                       | candidate                                                 | installed |
+-------------------------------------------------------------+-----------------------------------------------------------+-----------+
| java.package:org.eclipse.jdt.internal.compiler.apt.dispatch | org.eclipse.jdt.compiler.apt:1.4.300.v20221108-0856       | [x]       |
|                                                             | org.eclipse.jdt.core.compiler.batch:3.32.0.v20221108-1853 | [ ]       |
| java.package:org.eclipse.jdt.internal.compiler.apt.model    | org.eclipse.jdt.compiler.apt:1.4.300.v20221108-0856       | [x]       |
|                                                             | org.eclipse.jdt.core.compiler.batch:3.32.0.v20221108-1853 | [ ]       |
| java.package:org.eclipse.jdt.internal.compiler.apt.util     | org.eclipse.jdt.compiler.apt:1.4.300.v20221108-0856       | [x]       |
|                                                             | org.eclipse.jdt.core.compiler.batch:3.32.0.v20221108-1853 | [ ]       |
| java.package:org.eclipse.jdt.internal.compiler.tool         | org.eclipse.jdt.compiler.tool:1.3.200.v20220802-0458      | [x]       |
|                                                             | org.eclipse.jdt.core.compiler.batch:3.32.0.v20221108-1853 | [ ]       |
| java.package:org.osgi.service.log                           | org.eclipse.osgi:3.18.200.v20221116-1324                  | [x]       |
|                                                             | org.eclipse.osgi.services:3.11.100.v20221006-1531         | [ ]       |
| org.eclipse.equinox.p2.iu:org.eclipse.jdt.annotation        | org.eclipse.jdt.annotation:2.2.700.v20220826-1026         | [x]       |
|                                                             | org.eclipse.jdt.annotation:1.2.100.v20220826-1026         | [ ]       |
| org.eclipse.equinox.p2.iu:org.eclipse.jdt.annotation.source | org.eclipse.jdt.annotation.source:2.2.700.v20220826-1026  | [x]       |
|                                                             | org.eclipse.jdt.annotation.source:1.2.100.v20220826-1026  | [ ]       |
+-------------------------------------------------------------+-----------------------------------------------------------+-----------+
]


dev.equo.ide.gradle.P2MultitoolExamples._05[apt]=[
1 unit available with id org.eclipse.jdt.compiler.apt
  1.4.300.v20221108-0856  [x] included by install
+--------------------------------------------+-------------------------------------------------------------+
| key                                        | value                                                       |
+--------------------------------------------+-------------------------------------------------------------+
| id                                         | org.eclipse.jdt.compiler.apt                                |
| version                                    | 1.4.300.v20221108-0856                                      |
| maven coordinate                           | org.eclipse.jdt:org.eclipse.jdt.compiler.apt:1.4.300        |
| maven repo                                 | maven central?                                              |
| prop artifact-classifier                   | osgi.bundle                                                 |
| prop maven-artifactId                      | org.eclipse.jdt.compiler.apt                                |
| prop maven-groupId                         | org.eclipse.jdt                                             |
| prop maven-type                            | eclipse-plugin                                              |
| prop maven-version                         | 1.4.300-SNAPSHOT                                            |
| prop org.eclipse.equinox.p2.name           | Java Compiler Apt                                           |
| req org.eclipse.jdt.internal.compiler.tool | 2 available                                                 |
|                                            |   org.eclipse.jdt.compiler.tool:1.3.200.v20220802-0458      |
|                                            |   org.eclipse.jdt.core.compiler.batch:3.32.0.v20221108-1853 |
| req org.eclipse.jdt.core                   | org.eclipse.jdt.core:3.32.0.v20221108-1853                  |
+--------------------------------------------+-------------------------------------------------------------+
]


dev.equo.ide.gradle.P2MultitoolExamples._05[batch]=[
1 unit available with id org.eclipse.jdt.core.compiler.batch
  3.32.0.v20221108-1853  [ ] not included by install
+----------------------------------+-------------------------------------+
| key                              | value                               |
+----------------------------------+-------------------------------------+
| id                               | org.eclipse.jdt.core.compiler.batch |
| version                          | 3.32.0.v20221108-1853               |
| maven coordinate                 | org.eclipse.jdt:ecj:3.32.0          |
| maven repo                       | maven central?                      |
| prop artifact-classifier         | osgi.bundle                         |
| prop maven-artifactId            | org.eclipse.jdt.core                |
| prop maven-groupId               | org.eclipse.jdt                     |
| prop maven-type                  | eclipse-plugin                      |
| prop maven-version               | 3.32.0-SNAPSHOT                     |
| prop org.eclipse.equinox.p2.name | Eclipse Compiler for Java(TM)       |
+----------------------------------+-------------------------------------+
]


dev.equo.ide.gradle.P2MultitoolExamples._06[batch]=[
<unit id="org.eclipse.jdt.core.compiler.batch" singleton="false" version="3.32.0.v20221108-1853">
  <update id="org.eclipse.jdt.core.compiler.batch" range="[0.0.0,3.32.0.v20221108-1853)" severity="0"/>
  <properties size="7">
    <property name="org.eclipse.equinox.p2.name" value="Eclipse Compiler for Java(TM)"/>
    <property name="org.eclipse.equinox.p2.provider" value="Eclipse.org"/>
    <property name="maven-groupId" value="org.eclipse.jdt"/>
    <property name="maven-artifactId" value="org.eclipse.jdt.core"/>
    <property name="maven-version" value="3.32.0-SNAPSHOT"/>
    <property name="maven-classifier" value="batch-compiler"/>
    <property name="maven-type" value="eclipse-plugin"/>
  </properties>
  <provides size="25">
    <provided name="org.eclipse.jdt.core.compiler.batch" namespace="org.eclipse.equinox.p2.iu" version="3.32.0.v20221108-1853"/>
    <provided name="org.eclipse.jdt.core.compiler.batch" namespace="osgi.bundle" version="3.32.0.v20221108-1853"/>
    <provided name="org.eclipse.jdt.core" namespace="java.package" version="0.0.0"/>
    <provided name="org.eclipse.jdt.core.compiler" namespace="java.package" version="0.0.0"/>
    <provided name="org.eclipse.jdt.core.compiler.batch" namespace="java.package" version="0.0.0"/>
    <provided name="org.eclipse.jdt.internal.antadapter" namespace="java.package" version="0.0.0"/>
    <provided name="org.eclipse.jdt.internal.compiler" namespace="java.package" version="0.0.0"/>
    <provided name="org.eclipse.jdt.internal.compiler.apt.dispatch" namespace="java.package" version="0.0.0"/>
    <provided name="org.eclipse.jdt.internal.compiler.apt.model" namespace="java.package" version="0.0.0"/>
    <provided name="org.eclipse.jdt.internal.compiler.apt.util" namespace="java.package" version="0.0.0"/>
    <provided name="org.eclipse.jdt.internal.compiler.ast" namespace="java.package" version="0.0.0"/>
    <provided name="org.eclipse.jdt.internal.compiler.batch" namespace="java.package" version="0.0.0"/>
    <provided name="org.eclipse.jdt.internal.compiler.classfmt" namespace="java.package" version="0.0.0"/>
    <provided name="org.eclipse.jdt.internal.compiler.codegen" namespace="java.package" version="0.0.0"/>
    <provided name="org.eclipse.jdt.internal.compiler.env" namespace="java.package" version="0.0.0"/>
    <provided name="org.eclipse.jdt.internal.compiler.flow" namespace="java.package" version="0.0.0"/>
    <provided name="org.eclipse.jdt.internal.compiler.impl" namespace="java.package" version="0.0.0"/>
    <provided name="org.eclipse.jdt.internal.compiler.lookup" namespace="java.package" version="0.0.0"/>
    <provided name="org.eclipse.jdt.internal.compiler.parser" namespace="java.package" version="0.0.0"/>
    <provided name="org.eclipse.jdt.internal.compiler.parser.diagnose" namespace="java.package" version="0.0.0"/>
    <provided name="org.eclipse.jdt.internal.compiler.problem" namespace="java.package" version="0.0.0"/>
    <provided name="org.eclipse.jdt.internal.compiler.util" namespace="java.package" version="0.0.0"/>
    <provided name="org.eclipse.jdt.internal.compiler.tool" namespace="java.package" version="0.0.0"/>
    <provided name="org.eclipse.jdt.core.compiler.batch" namespace="osgi.identity" version="3.32.0.v20221108-1853">
      <properties size="1">
        <property name="type" value="osgi.bundle"/>
      </properties>
    </provided>
    <provided name="bundle" namespace="org.eclipse.equinox.p2.eclipse.type" version="1.0.0"/>
  </provides>
  <artifacts size="1">
    <artifact classifier="osgi.bundle" id="org.eclipse.jdt.core.compiler.batch" version="3.32.0.v20221108-1853"/>
  </artifacts>
  <touchpoint id="org.eclipse.equinox.p2.osgi" version="1.0.0"/>
  <touchpointData size="1">
    <instructions size="1">
      <instruction key="manifest">
            Bundle-SymbolicName: org.eclipse.jdt.core.compiler.batch
Bundle-Version: 3.32.0.v20221108-1853
          </instruction>
    </instructions>
  </touchpointData>
</unit>
]